drivers/sensor/ms5837/ms5837.c

```diff
@@ -285,6 +285,57 @@ static int ms5837_read_prom(const struct device *dev, const uint8_t cmd,
 	return 0;
 }
 
+static uint8_t crc4(uint16_t n_prom[8])
+{
+	uint16_t n_rem = 0;
+	uint8_t n_bit;
+
+	n_prom[0] = ((n_prom[0]) & 0x0FFF);
+	n_prom[7] = 0;
+
+	for (int cnt = 0; cnt < 16; cnt++) {
+		if (cnt % 2 == 1) {
+			n_rem ^= (unsigned short) ((n_prom[cnt>>1]) & 0x00FF);
+		}  else {
+			n_rem ^= (unsigned short) (n_prom[cnt>>1]>>8);
+		}
+
+		for (n_bit = 8; n_bit > 0; n_bit--) {
+			if (n_rem & (0x8000)) {
+				n_rem = (n_rem << 1) ^ 0x3000;
+			} else {
+				n_rem = (n_rem << 1);
+			}
+		}
+	}
+	n_rem = ((n_rem >> 12) & 0x000F);
+	/* final 4-bit remainder is CRC code */
+	return (n_rem ^ 0x00);
+}
```

Comment:
Please move this to live with the other crc implementations in `lib/os`

Reply:
Thanks for reviewing. I don't think I have capacity at the moment to move the crc and test it all... If it is really necessary then I'd prefer to just drop the last commit of this series.
I actually prefer having it like this, as it is basically a copy of the reference code in the datasheet, so super easy to check against. Writing a generic crc4 func and then all the prom contents manipulation doesn't feel like it would simplify.
Would you prefer if the func was renamed to something like "calc_prom_crc"?

Reply:
I frequently see PRs that try to implement their own crc functions when they could instead use the shared implementations. I suggest dropping the last commit for now and revisiting the crc issue in a follow up PR when you have time.